client/src/pages/Profile/ProfilePage.jsx

```diff
@@ -13,14 +13,25 @@ const ProfilePage = () => {
   useEffect(() => {
     const fetchData = async () => {
       try {
-        const response = await fetch("http://localhost:8080/api/get-workouts");
-        const responseData = await response.json();
-        setData(responseData);
-        // todo: until getUserById is implemented, we will hard code the user to be accountId=1
-        const responseProfile = responseData.filter(
-          (data) => data.accountId !== 1
-        );
-        setProfileData(responseProfile[0]);
+        const storedToken = sessionStorage.getItem("token");
+        if (storedToken) {
+          const response = await fetch(
+            "http://localhost:8080/api/get-workouts",
+            {
+              headers: {
+                Authorization: `Bearer ${storedToken}`,
```

Comment:
LGTM but I think setting cookies (from the server) should be simpler since you don't have to send an auth token to every request?